common/src/main/java/com/microsoft/identity/common/internal/controllers/BrokerMsalController.java

```diff
@@ -1600,4 +1693,148 @@ private void verifyBrokerVersionIsSupported(@Nullable final Bundle resultBundle,
+     */
+    @NonNull
+    private BrokerInteractiveTokenCommandParameters buildInteractiveTokenParametersForWebApps(@NonNull final WebAppsGetTokenSubOperationRequest webAppsRequest,
+                                                                                              @NonNull final WebAppsAdditionalRequiredParameters requiredParams,
+                                                                                              @NonNull final String minBrokerProtocolVersion) throws BaseException {
+        final String authorityUrl = webAppsRequest.getAuthority() != null
+                ? webAppsRequest.getAuthority()
+                : WebAppsGetTokenSubOperationRequest.DEFAULT_AUTHORITY;
+
+        if (StringUtil.isNullOrEmpty(authorityUrl)) {
+            throw new ClientException(ClientException.MISSING_PARAMETER, "Authority is null or empty.");
+        }
+        final String clientId = webAppsRequest.getClientId();
+        if (StringUtil.isNullOrEmpty(clientId)) {
+            throw new ClientException(ClientException.MISSING_PARAMETER, "ClientId is null or empty.");
+        }
+        final String redirect = webAppsRequest.getRedirectUri();
+        if (StringUtil.isNullOrEmpty(redirect)) {
+            throw new ClientException(ClientException.MISSING_PARAMETER, "Redirect URI is null or empty.");
+        }
+
+        final String correlationId = webAppsRequest.getCorrelationId() != null
+                ? webAppsRequest.getCorrelationId()
+                : UUID.randomUUID().toString();
+
+        final Set<String> scopeSet;
+        final String rawScope = webAppsRequest.getScopes();
+        if (StringUtil.isNullOrEmpty(rawScope)) {
+            scopeSet = Collections.emptySet();
+        } else {
+            scopeSet = new HashSet<>();
+            for (String s : rawScope.trim().split("\\s+")) {
+                if (!s.isEmpty()) {
+                    scopeSet.add(s);
+                }
+            }
+        }
+
+        final Authority authority = Authority.getAuthorityFromAuthorityUrl(authorityUrl);
+        if (authority == null) {
+            throw new ClientException(ClientException.MISSING_PARAMETER,
+                    "Unable to create Authority from url");
+        }
+        if (authority instanceof AzureActiveDirectoryAuthority) {
+            ((AzureActiveDirectoryAuthority) authority)
+                    .setMultipleCloudsSupported(webAppsRequest.getInstanceAware());
+        }
+
+        Map<String, String> tempMap = webAppsRequest.getExtraParameters();
+        final Map<String, String> extraQueryParamsMap = tempMap != null
+                ? tempMap
+                : Collections.emptyMap();
+        final List<Map.Entry<String, String>> queryParams = new ArrayList<>(extraQueryParamsMap.size() + 1);
+        for (Map.Entry<String, String> e : extraQueryParamsMap.entrySet()) {
+            queryParams.add(new java.util.AbstractMap.SimpleEntry<>(e.getKey(), e.getValue()));
+        }
+
```

Comment:
simplify to

 final List<Map.Entry<String, String>> queryParams = new ArrayList<>(extraQueryParamsMap.entrySet());